### PR TITLE
online-book-store/dto-web

### PR DESCRIPTION
**Pull request by bodya4243**
1) create Dto
2) create controller
3) create mapper
4) create mapper config
5) create exception

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/dto/BookDto.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    private Long id;
+    
+    @Column(name = "title", nullable = false)
+    private String title;
+    
+    @Column(name = "author", nullable = false)
+    private String author;
+    
+    @Column(name = "price", nullable = false)
+    private BigDecimal price;
```
Comment (slizko1) [inline src/main/java/com/example/onlinebookstore/dto/BookDto.java]: Why do we need hibernate annotations for dto class? Do we need tosave them in the database?

File: src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+public class CreateBookRequestDto {
+    @Column(name = "title", nullable = false)
+    private String title;
+    
+    @Column(name = "author", nullable = false)
+    private String author;
+    
+    @Column(name = "price", nullable = false)
+    private BigDecimal price;
+}
```
Comment (slizko1) [inline src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java]: Same here

File: src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -10,14 +14,26 @@
...
+    
+    private final BookMapper bookMapper;
+    
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto save(CreateBookRequestDto bookDto) {
+        Book book = bookMapper.toModel(bookDto);
+        book.setDescription("new description");
+        book.setIsbn("isbn");
```
Comment (slizko1) [inline src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java]: Post request will look like this фacording to the assignment guideline`{
  "title": "Sample Book 3",
  "author": "Author C",
  "isbn": "9781122334455",
  "price": 29.99,
  "description": "Yet another sample book description.",
  "coverImage": "http://example.com/cover3.jpg"
}
` So you don't neen to set this fields

File: pom.xml [hunk 4]
```diff
@@ -76,8 +78,47 @@
...
+							<version>${mapstruct.version}</version>
+						</path>
+					</annotationProcessorPaths>
 				</configuration>
 			</plugin>
 		</plugins>
 	</build>
-</project>
+
+</project>
```
Comment (slizko1) [inline pom.xml]: Add empty line

**Review by slizko1 — COMMENTED** [review]
Gj! But some fixes are needed

File: pom.xml [hunk 5]
```diff
@@ -5,60 +5,62 @@
 	<parent>
 		<groupId>org.springframework.boot</groupId>
 		<artifactId>spring-boot-starter-parent</artifactId>
-		<version>3.2.5</version>
-		<relativePath/>
+		<version>3.2.4</version>
+		<relativePath/> <!-- lookup parent from repository -->
```
Comment (andrii-hoienko) [inline pom.xml:9]:
```suggestion
		<relativePath/>
```

File: pom.xml (hunk 4 above)
Comment (andrii-hoienko) [reply pom.xml]: Not fixed

File: src/main/java/com/example/onlinebookstore/dto/BookDto.java (hunk 1 above)
Comment (andrii-hoienko) [reply src/main/java/com/example/onlinebookstore/dto/BookDto.java]: Agree

File: src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (andrii-hoienko) [reply src/main/java/com/example/onlinebookstore/dto/CreateBookRequestDto.java]: Agree

File: src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java (hunk 3 above)
Comment (andrii-hoienko) [reply src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java]: agree

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Please read students comments

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**